hc5/src/main/java/feign/hc5/AsyncApacheHttp5Client.java

```diff
@@ -52,6 +53,8 @@ public final class AsyncApacheHttp5Client implements AsyncClient<HttpClientConte
 
   private final CloseableHttpAsyncClient client;
 
+  protected volatile BiFunction<Request, CompletableFuture<Response>, FutureCallback<SimpleHttpResponse>> futureCallbackFactory;
```

Comment:
I'm not against having a factory, but it should be something final that you set and forget...

if you need different handling on different clients, you use different builders for it....

So, can we make it final and include as part of the constructor and use current value when none is present?!